core/authenticate/service.go

```diff
@@ -316,6 +375,207 @@ func (s Service) applyMailOTP(ctx context.Context, request RegistrationFinishReq
 	}, nil
 }
 
+func (s Service) startPassKeyRegisterMethod(ctx context.Context, newPassKeyUser *strategy.UserData, flow *Flow) (*RegistrationStartResponse, error) {
+	options, session, err := s.web.BeginRegistration(newPassKeyUser)
+	if err != nil {
+		return &RegistrationStartResponse{}, err
+	}
+
+	session.Challenge = base64.RawURLEncoding.EncodeToString([]byte(session.Challenge))
+	sessionInBytes, err := json.Marshal(session)
+	if err != nil {
+		return &RegistrationStartResponse{}, err
+	}
+	flow.Metadata["passkey_session"] = sessionInBytes
```

Comment:
store here type of session as well, like register/login

Reply:
done